### PR TITLE
fixtures: deprecate pytest.yield_fixture()

### PR DESCRIPTION
**Pull request by bluetech**

### CODE REVIEW

File: src/_pytest/deprecated.py [hunk 1]
```diff
@@ -32,6 +32,9 @@
     "Please update to the new name.",
 )
 
+YIELD_FIXTURE = PytestDeprecationWarning(
+    "@pytest.yield_fixture is deprecated.\nUse @pytest.fixture instead."
```
Comment (nicoddemus) [inline src/_pytest/deprecated.py:36]: The bit about being safe to search/replace might be useful here. 😁

**Review by nicoddemus — APPROVED** [review]

File: testing/python/fixtures.py [hunk 2]
```diff
@@ -3636,10 +3620,10 @@ def test_1(arg1):
         """
         )
 
-    def test_yields_more_than_one(self, testdir, flavor):
+    def test_yields_more_than_one(self, testdir: Testdir) -> None:
         testdir.makepyfile(
             """
-            from test_context import fixture
+            import pytest
             @fixture(scope="module")
```
Comment (The-Compiler) [inline testing/python/fixtures.py:3627]: Missed one :wink:

```suggestion
            @pytest.fixture(scope="module")
```

**Review by The-Compiler — CHANGES_REQUESTED** [review]
Nice! I didn't realize it wasn't deprecated already... Looks like it's [widely used](https://github.com/search?q=yield_fixture&type=code) but I'm still +1 for getting rid of it.

File: src/_pytest/deprecated.py (hunk 1 above)
Comment (bluetech) [reply src/_pytest/deprecated.py]: Trying to keep it short, I added "they are the same".

File: testing/python/fixtures.py (hunk 2 above)
Comment (bluetech) [reply testing/python/fixtures.py]: Oops, fixed, thanks.

**Review by asottile — APPROVED** [review]

**Review by nicoddemus — APPROVED** [review]

**Review by The-Compiler — APPROVED** [review]

**Comment by The-Compiler** [comment]
Looks like this now conflicts with #7985 which I just merged, sorry!

**Pull request merged**